### ISSUE TITLE
NoSuchMethodException when using redisson client in Quarkus native mode

### ISSUE THREAD

**Issue by denysandriyanov** [redisson/redisson#4811]
Hello, when using redisson in quarkus native mode, application fails with following.
Please help

{"timestamp":"2023-01-22T16:08:52.199Z","level":"ERROR","message":"Failed to start application (with profile prod)","mdc":{},"exception":{"refId":1,"exceptionType":"java.lang.NoSuchMethodException","message":"com.esotericsoftware.kryo.serializers.TimeSerializers$OffsetDateTimeSerializer.<init>()","frames":[{"class":"java.lang.Class","method":"getConstructor0","line":3585},{"class":"java.lang.Class","method":"newInstance","line":626},{"class":"com.esotericsoftware.kryo.SerializerFactory$ReflectionSerializerFactory","method":"newSerializer","line":80},{"class":"com.esotericsoftware.kryo.SerializerFactory$ReflectionSerializerFactory","method":"newSerializer","line":64},{"class":"com.esotericsoftware.kryo.Kryo","method":"getDefaultSerializer","line":451},{"class":"com.esotericsoftware.kryo.util.DefaultClassResolver","method":"registerImplicit","line":89},{"class":"com.esotericsoftware.kryo.Kryo","method":"getRegistration","line":581},{"class":"com.esotericsoftware.kryo.Kryo","method":"writeObjectOrNull","line":654},{"class":"com.esotericsoftware.kryo.serializers.RecordSerializer","method":"write","line":105},{"class":"com.esotericsoftware.kryo.Kryo","method":"writeClassAndObject","line":711},{"class":"org.redisson.codec.Kryo5Codec$5","method":"encode","line":125},{"class":"org.redisson.command.CommandAsyncService","method":"encode","line":622},{"class":"org.redisson.RedissonObject","method":"encode","line":313},{"class":"org.redisson.RedissonBucket","method":"setAsync","line":178},{"class":"org.redisson.RedissonBucket","method":"set","line":169},{"class":"com.avid.configurator.service.service.redis.RegistrationService","method":"setRegistrationData","line":95}

**Comment by mrniko** [comment]
as temporary workaround add line below to reflect-config.json

```
{
  "name": "com.esotericsoftware.kryo.serializers.TimeSerializers$OffsetDateTimeSerializer",
  "allDeclaredConstructors":true
}
```

**Comment by mrniko** [comment]
Fixed! Thanks for report

**Comment by denysandriyanov** [comment]
> as temporary workaround add line below to reflect-config.json
> 
> ```
> {
>   "name": "com.esotericsoftware.kryo.serializers.TimeSerializers$OffsetDateTimeSerializer",
>   "allDeclaredConstructors":true
> }
> ```

Hello. I tried this, unfortunately did not help

**Comment by denysandriyanov** [comment]
As a side note it's not only this serializer, it's for any serialyzer. for example enum

	at org.redisson.command.CommandAsyncService.encode(CommandAsyncService.java:622)
	at org.redisson.RedissonObject.encode(RedissonObject.java:313)
	at org.redisson.RedissonBucket.setIfAbsentAsync(RedissonBucket.java:246)
	at org.redisson.RedissonBucket.setIfAbsent(RedissonBucket.java:229)
	at com.avid.configurator.service.service.redis.NotificationConsumer.notificationIsNotHandled(NotificationConsumer.java:82)
	at com.avid.configurator.service.service.redis.NotificationConsumer.onMessage(NotificationConsumer.java:56)
	at com.avid.configurator.service.service.redis.NotificationConsumer.onMessage(NotificationConsumer.java:27)
	at com.avid.configurator.service.service.redis.NotificationConsumer_ClientProxy.onMessage(Unknown Source)
	at org.redisson.PubSubMessageListener.onMessage(PubSubMessageListener.java:79)
	at org.redisson.client.RedisPubSubConnection.onMessage(RedisPubSubConnection.java:68)
	at org.redisson.client.handler.CommandPubSubDecoder$1.run(CommandPubSubDecoder.java:164)
	at java.base@17.0.5/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base@17.0.5/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@17.0.5/java.lang.Thread.run(Thread.java:833)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
Caused by: java.lang.InstantiationException: com.esotericsoftware.kryo.serializers.DefaultSerializers$EnumSerializer
	at java.base@17.0.5/java.lang.Class.newInstance(DynamicHub.java:639)
	at com.esotericsoftware.kryo.SerializerFactory$ReflectionSerializerFactory.newSerializer(SerializerFactory.java:80)
	... 25 more
Caused by: java.lang.NoSuchMethodException: com.esotericsoftware.kryo.serializers.DefaultSerializers$EnumSerializer.<init>()
	at java.base@17.0.5/java.lang.Class.getConstructor0(DynamicHub.java:3585)
	at java.base@17.0.5/java.lang.Class.newInstance(DynamicHub.java:626)
	... 26 more

**Comment by mrniko** [comment]
Can you share the object you trying to serialize?

**Comment by denysandriyanov** [comment]
public enum Status {
    IN_PROGRESS,
    FAILED,
    COMPLETED;

    @JsonCreator
    public static Status from(String status) {
        return Stream.of(values()).filter(v -> v.name().equalsIgnoreCase(status)).findFirst()
                .orElseThrow(() ->
                {
                    Log.error(String.format("Invalid status: %s" +
                            "\nValid values: INIT, FAILED, COMPLETED", status));

                    return new EnumConstantNotPresentException(Status.class, status);
                });
    }

    @JsonValue
    public String getValue() {
        return name();
    }
}

**Comment by mrniko** [comment]
Fixed! All Serializers have been added to reflect-config.json

**Comment by denysandriyanov** [comment]
Great, thank you. Could you please update when this fixes are release? it will be included in 3.19.2?

**Comment by mrniko** [comment]
3.19.2 is scheduled for release in early February.
